src/main/java/edu/cornell/kfs/concur/rest/resource/ConcurAIResource.java

```diff
@@ -0,0 +1,99 @@
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.service.AccountService;
+import org.kuali.kfs.datadictionary.legacy.DataDictionaryService;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+
+import com.google.gson.Gson;
+
+import edu.cornell.kfs.concur.ConcurConstants.ConcurAIConstants;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurAccountDetailDto;
+
+@Path("api")
+@Produces(MediaType.APPLICATION_JSON)
+public class ConcurAIResource {
+    private static final Logger LOG = LogManager.getLogger();
+    private static final Gson gson = new Gson();
+    
+    private AccountService accountService;
+    private DataDictionaryService dataDictionaryService;
+    
+    @Context
+    protected HttpServletRequest servletRequest;
+
+    @Context
+    protected HttpServletResponse servletResponse;
+    
+    @GET
+    public Response describeConcurAIResource() {
+        return Response.ok(ConcurAIConstants.RESOURCE_NAME).build();
+    }
+    
+    @GET
+    @Path("/getAccountDetails")
+    public Response getAccountDetails() {
+        try {
+            String chart = servletRequest.getParameter(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE);
+            String accountNumber = servletRequest.getParameter(KFSPropertyConstants.ACCOUNT_NUMBER);
+            LOG.debug("getAccountDetails, entering with chart {} and account {}", chart, accountNumber);
+            
+            if (!validatePropertyValue(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, chart) || 
+                    !validatePropertyValue(KFSPropertyConstants.ACCOUNT_NUMBER, accountNumber)) {
+                LOG.debug("getAccountDetails, account or chart invalid");
+                return Response.status(Status.BAD_REQUEST).entity(ConcurAIConstants.CHART_AND_ACCOUNT_MUST_BE_PROVIDED).build();
+            }
+            
+            Account account = getAccountService().getByPrimaryId(chart, accountNumber);
+            
+            if (account != null) {
+                ConcurAccountDetailDto dto = new ConcurAccountDetailDto(account);
+                return Response.ok(gson.toJson(dto)).build();
+            } else {
+                String notFoundMessage = MessageFormat.format(ConcurAIConstants.ACCOUNT_NOT_FOUND_MESSAGE, chart, accountNumber);
+                return Response.status(Status.NOT_FOUND).entity(notFoundMessage).build();
+            }
+            
+        } catch (Exception e) {
+            LOG.error("getAccountDetails, had an error getting account details", e);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity(ConcurAIConstants.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+    
+    private boolean validatePropertyValue(String propertyName, String propertyValue) {
```

Comment:
Since this is only validating Account properties should the method name be updated to reflect that? or should the class be added to method params?

Reply:
I renamed it for account properties